src/Shared/HttpSys/RequestProcessing/RequestHeaders.cs

```diff
@@ -97,7 +110,12 @@ public ICollection<string> Keys
 
     public int Count
     {
-        get { return PropertiesKeys().Count() + Extra.Count; }
+        get
+        {
+            int count = GetKnownHeadersCount(HeaderKeys);
```

Comment:
HeaderKeys doesn't need to be passed in here either

Reply:
ack